cpg_workflows/jobs/picard.py

```diff
@@ -18,6 +18,48 @@
 from cpg_workflows.utils import can_reuse, exists
 
 
+def blacklist_intervals(
+    b: hb.Batch,
+    job_attrs: dict | None = None,
+    source_intervals_path: Path | None = None,
+    blacklist_intervals_path: Path | None = None,
+) -> Job | None:
+    """
+    Add a job that subtracts the blacklisted intervals from the source intervals.
+
+    @param b: Hail Batch object,
+    @param job_attrs: attributes for Hail Batch job,
+    @param source_intervals_path: path to source intervals to use in joint calling.
+        Would check for config if not provided.
+    @param blacklist_intervals_path: path to blacklist intervals to exclude. 
+        Would check for config if not provided.
+
+    The job calls picard IntervalListTools to subtract the blacklist interval list
+    from the source interval list.
+    """
+    sequencing_type = get_config()['workflow']['sequencing_type']
+    source_intervals_path = source_intervals_path or reference_path(f'broad/{sequencing_type}_calling_interval_lists')
+    blacklist_intervals_path = blacklist_intervals_path or reference_path('broad/blacklist_intervals') #TODO add this to config (default?)
+    
+    j = b.new_job(
+        f'Subtract blacklist intervals before scattering intervals for {sequencing_type} calling',
+        attributes=(job_attrs or {}) | dict(tool='picard IntervalListTools'),
+    )
+    j.image(image_path('picard'))
+    STANDARD.set_resources(j, storage_gb=4, mem_gb=2)
+    
+    cmd = f"""
+    mkdir $BATCH_TMPDIR/out
+
+    picard -Xms1000m -Xmx1500m \
+    IntervalListTools \
+    ACTION=SUBTRACT \
+    INPUT={b.read_input(str(source_intervals_path))} \
+    SECOND_INPUT={b.read_input(str(blacklist_intervals_path))} \
+    OUTPUT=$BATCH_TMPDIR/out/subtracted.interval_list
```

Comment:
this will work for the batch, but might be more future proof if we create an output path incorporating the values of the source and subtracted intervals path - save that into long term storage. That way we've got a permanent record of the A-B result, and future runs with the same intervals won't need to recreate this

Reply:
Yeah good thought. If we can create it once and nicely document that it's the standard `wgs_calling_regions.hg38.interval_list` intervals list with the `telomeres_and_centromeres` list subtracted from it then we can just reference it directly without deriving it each time a batch is run.

Reply:
I still fully believe that these interval lists should be stored in long term memory, but looking at it now, the output prefixes are all in temp buckets anyway. I'll make an issue, but no reason to hold back this change